tests/bson/bson-int64-operation-003.phpt

```diff
@@ -0,0 +1,43 @@
+--TEST--
+MongoDB\BSON\Int64 operations: increment/decrement
+--FILE--
+<?php
+
+$value = new MongoDB\BSON\Int64(10);
+
+var_dump($value++); // This will print 11 instead of 10 due to PHP not implementing operator overloading for increment
```

Comment:
Consider a comment above all of this explaining that inc/dec operations _do not_ modify the object itself, just so we have that documented somewhere.

I remember we talked about that behavior, but I was reminded when reading the test wondering why `$value` always seems to reset to 10 for each operation.

Reply:
Quick follow-up, as I just saw [this reply in room11](https://chat.stackoverflow.com/transcript/message/56345198#56345198) to your question about inc/dec operations on objects:

> PHP allows you to overload operators in any way you want, thus you're responsible to make a new clone for the result - hopefully Int64 is already immutable. I've learned how operator overloading is working from [gmp implementation](https://github.com/php/php-src/blob/8d0345d94ef98bfe29df264b338aeef16ddefda9/ext/gmp/gmp.c#L410C20-L427), as you can see there - if result and op1 are the same, gmp makes clone of op1.

It looks like the `do_operation` handler always initializes a new `result` zval, but I can't be sure. Can you add a test or two to assert that Int64 objects are immutable?

I suppose that could be accomplished with something like `$value !== $value++`.